### PR TITLE
Removed hacks and described task-api-dev mode instead

### PR DESCRIPTION
**Pull request by maaktweluit**
Updates to the docs for the changes merged in golemfactory/golem#5046

@jamuszyn:
- Please do not update the public docs until release 0.23 is out
- There will be another comfort feature added that changes this guide, would recommend updating the task-api-docs after.

### CODE REVIEW

File: tutorial.md [hunk 1]
```diff
@@ -850,10 +836,9 @@ python scripts/task_api_tests/basic_integration.py task-from-app-def ../tutorial
 To run the tutorial app between your own nodes:
 - Add the app descriptor JSON file to your requestor's data-dir
 - Build the docker image on all nodes
-- Whitelist the image repository on both requestor and provider side.
-  _TODO: add task-api-dev config to not pull images and whitelist local images_
 - create a `task.json` file to request a task
 - Enter a private network ( optional, recommended to not have others pick up your task )
+- Run golem with task-api developer mode enabled
```
Comment (etam) [inline tutorial.md]: At this point it's unknown what is a developer mode.

**Review by etam — APPROVED** [review]

File: tutorial.md [hunk 2]
```diff
@@ -896,10 +879,14 @@ Update your `golem.db` in the table `appconfig` app with ID `801964d531675a235c9
...
     }
 ```
-5. (optional) Enter a private network by starting all golem nodes with the `protocol_id` option
+4. (optional) Enter a private network by starting all golem nodes with the `protocol_id` option
 ```bash
 golemapp --protocol_id=1337
 ```
+5. Start both golem nodes with the task-api developer mode enabled
+```bash
+golemapp --task-api-dev=1
```
Comment (mdtanrikulu) [inline tutorial.md:888]: When I try this command, terminal gives;
```
Error: --task-api-dev option does not take a value
```

So, seemingly this should be  `golemapp --task-api-dev`

Comment (maaktweluit) [reply tutorial.md:888]: Thanks for the heads up! we continued working on this in golemfactory/task-api-docs
Checked the argument and its proper there :)

Closing this PR and will make a new one syncing the 2 docs

**Comment by maaktweluit** [comment]
Closing this PR, progress has been made in golemfactory/task-api-docs

**Pull request closed without merging**